docs/cli-reference.md

```diff
@@ -1 +1,228 @@
-# CLI Reference
\ No newline at end of file
+# CLI Reference
+Dozer Command Line Interface is a unified tool and exposes commands for publishing and managing your applications, as well as lower level tools to interact with each component of your application. This reference guides you through getting started with the CLI.
+
+## Synopsis
+```
+dozer [options] <command> <subcommand> [parameters]
+```
+Use `dozer help`, `dozer -h` or `dozer --help` to view a list of available commands. Use `dozer <command> help`, `dozer <command> -h` or `dozer <command> --help` for information on a specific command with available subcommands. The synopsis for each command shows its parameters and their usage. Optional parameters are shown in square brackets.
+
+## Development Flow
+
+![Dozer CLI Flow](@site/static/docs/cli.svg)
+
+## Local Development
+
+### `dozer init`
+If you like to initialize dozer application, dozer will create `dozer-config.yaml` in the workspace with pre-populated configuration template.
+
+> ```bash
+> question: App name (quick-start-app): 
+> question: Home directory (./.dozer): 
+> question: Connection Type - one of: [P]ostgres, [E]thereum, [S]nowflake (Postgres): 
+> question: Config path (./dozer-config.yaml): 
+> ```
+
+### `dozer live`
+Run following command to edit code for your dozer application.
+
+> ```bash
+> .____   ___ __________ ____
+> |  _ \ / _ \__  / ____|  _ \
+> | | | | | | |/ /|  _| | |_) |
+> | |_| | |_| / /_| |___|  _ <
+> |____/ \___/____|_____|_| \_\
+> 
+> INFO Starting live server
+> ```
+
+### `dozer`
+If you run Dozer CLI with no command are passed like below, dozer will bring up both `app` and `api` services.
+```bash
+dozer
+# or 
+# dozer -c <custom-dozer-config-path>
+```
+
+## Self-Hosted Deployment
+
+### `dozer build`
+Run following command to initialize and lock the schema definitions. Once initialized, schemas cannot be changed.
+
+> ```bash
+>  INFO Initiating app: ..
+>  ...
+>  INFO Created new build v0001
+> ```
+
+### `dozer run app`
+If you run Dozer CLI with `run` command with `app` subcommand like below, dozer will bring up app service.
+
+### `dozer run api`
+If you run Dozer CLI with `run` command with `api` subcommand like below, dozer will bring up api service.
+
```

Comment:
Add ..for more details on self-hosting click here. And we link to the deployment section